### ISSUE TITLE
multi-entity endpoint

### ISSUE THREAD

**Issue by oneilsh** [monarch-initiative/monarch-app#270]
I'm looking for a blunt-instrument API endpoint to support the R package. What I'd like to do is provide a list of entities, and a list of entity categories that they might connect to, and get those associations back in bulk.

Here's an example request:

```yaml
entities:
  - MONDO:103135
  - HGNC:1356
categories:
  - biolink:Gene
  - biolink:PhenotypicQuality
limit_per_group: 10
offset_per_group: 0
``` 

And an example result:

```yaml
entities:
    - id: "MONDO:103135":
      name: "Some Rare Syndrome"
      category: "biolink:Disease"
      associated_entity_categories:
            - category: "biolink:Gene":
              total: 140
              associations:
                   - id: "HGNC:13513"
                     name: "HJPR1"
                     predicate: "gene_correlated_with"
                     association_category: "biolink:CorrelatedGeneToDiseaseAssociation"
                     publications: ...
                   - id: "HGNC:161615"
                     name: "LN3"
                     predicate: "gene_causes"
                     publications: ...
                     association_category: "biolink:CausalGeneToDiseaseAssociation":
            - category: "biolink:PhenotypicQuality":
              total: 30
              associations:
                   - id: "HP:13951134"
                     name: "Short stature"
                     predicate: "phenotype_associated_with"
                     publications: ...
                     association_category: "biolink:DiseaseToPhenotypicFeatureAssociation"
                   - ...
                   - ...
    - id: "HGNC:1356":
      category: "biolink:Gene"
      connected_entity_categories:
           - category: "biolink:Gene":
             total: 8
             associations:
                  - id: "HGNC:1352"
                    name: "JJNS2"
                    predicate: "interacts_with"
                    publications: ...
                    association_category: "biolink:GeneGeneInteractionAssociation"
                   - ...
                   - ...
           - category: "biolink:PhenotypicQuality":
             total: 5
             associations:
             ...
```

The limit and offset would be applied per combination, so the maximum number of results should be `#entities * #categories * limit_per_group`. If necessary the total number of entities, categories, and limit can be limited to keep performance. As the offset increases it's expected that some groups will run out of results before others.

**Comment by kevinschaper** [comment]
I had been thinking about this in terms of using solr's result grouping, and churning a little on how I could hack multi-level grouping, when solr only supports grouping by a single field... but maybe it makes more sense to just build this response by doing a separate solr query for each of the permutations of entities and categories specified. 

That means that this functionally this just wraps the association endpoint call, and we just need to a be a little protective of memory - maybe with some kind of sensible limiting on the number of entities.

**Comment by glass-ships** [comment]
Does it make sense for `limit_per_group` to refer to a limit per entity, or per category?

**Comment by oneilsh** [comment]
I think per-entity+category; for example if the entity is a disease and we are searching for counterpart categories genes and phenotypes, if there are 100 related genes and 3 related phenotypes, we don't want to have to page through all of the genes to get to the top phenotypes

**Comment by glass-ships** [comment]
Resolved in #310
